crates/router/src/connector/adyen.rs

```diff
@@ -1910,3 +1910,324 @@ impl api::IncomingWebhook for Adyen {
+        header.append(&mut api_key);
+        Ok(header)
+    }
+
+    fn get_url(
+        &self,
+        req: &types::SubmitEvidenceRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let endpoint = build_env_specific_endpoint(
+            connectors.adyen.third_base_url.as_str(),
+            req.test_mode,
+            &req.connector_meta_data,
+        )?;
+        Ok(format!("{}supplyDefenseDocument", endpoint))
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::SubmitEvidenceRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
+        let merchant_account_code = get_key(&req.connector_auth_type)?;
+        let connector_req = adyen::Evidence::try_from((req, merchant_account_code))?;
+        Ok(RequestContent::Json(Box::new(connector_req)))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::SubmitEvidenceRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        let request = services::RequestBuilder::new()
+            .method(services::Method::Post)
+            .url(&types::SubmitEvidenceType::get_url(self, req, connectors)?)
+            .attach_default_headers()
+            .headers(types::SubmitEvidenceType::get_headers(
+                self, req, connectors,
+            )?)
+            .set_body(types::SubmitEvidenceType::get_request_body(
+                self, req, connectors,
+            )?)
+            .build();
+        Ok(Some(request))
+    }
+
+    fn handle_response(
+        &self,
+        data: &types::SubmitEvidenceRouterData,
+        _event_builder: Option<&mut ConnectorEvent>,
+        _res: types::Response,
+    ) -> CustomResult<types::SubmitEvidenceRouterData, errors::ConnectorError> {
+        Ok(types::SubmitEvidenceRouterData {
+            response: Ok(types::SubmitEvidenceResponse {
+                dispute_status: api_models::enums::DisputeStatus::DisputeChallenged,
+                connector_status: None,
+            }),
+            ..data.clone()
+        })
+    }
+
+    fn get_error_response(
+        &self,
+        res: types::Response,
+        event_builder: Option<&mut ConnectorEvent>,
+    ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
+        self.build_error_response(res, event_builder)
+    }
+}
+impl api::UploadFile for Adyen {}
+impl api::RetrieveFile for Adyen {}
+impl
+    services::ConnectorIntegration<
+        api::Retrieve,
+        types::RetrieveFileRequestData,
+        types::RetrieveFileResponse,
+    > for Adyen
+{
+}
+impl
+    services::ConnectorIntegration<
+        api::Upload,
+        types::UploadFileRequestData,
+        types::UploadFileResponse,
+    > for Adyen
+{
+}
+#[async_trait::async_trait]
+impl api::FileUpload for Adyen {
+    fn validate_file_upload(
```

Comment:
Can you please add validations specific to adyen connector, https://help.adyen.com/knowledge/risk/dispute-management/what-are-the-specific-defense-requirements